data/tutorials/language/0lg_08_env.md

```diff
@@ -0,0 +1,557 @@
+
+Since, by definition, effects lie outside function types, a function type can't reflect the effects a function may have. However, it is important to document the intended side effects a function may have. Consider the `Unix.time` function. It returns the number of seconds elapsed since Jan 1, 1970.
+```ocaml
+# Unix.time ;;
+- : unit -> float = <fun>
+```
+
+This function does not need to be passed any data to return its result. It is entirely context-dependent, entirely determined by external factors. Therefore, it could be passed anything, which would simply be discarded. Note it has to be passed something to trigger the computation, which is requesting the information to the operating system, as suggested by the module name. Since that function needs some data to trigger the computation but the data is going to be ignored, it makes sense to provide `()`, the unit value, the inhabitant of the type which only has a single value.
+
+A similar reasoning applies to functions producing an effect instead of being effect-determined functions. Consider `print_endline`. It prints the string it was passed to standard output, followed by a line termination.
+```ocaml
+# print_endline;;
+- : string -> unit = <fun>
+```
+Since the purpose of the function is only to produce an effect, it has no meaningful data to return; therefore, again, it makes sense to return the unit value.
+
+This illustrates the relationship between functions intended to have side effects and the unit type. The presence of the unit type does not indicate the presence of side effects. The absence of the unit type does not indicate the absence of side effects. But when no data needs to be passed as input or can be returned as output, the unit type should be used to indicate it and suggest the presence of side effects.
+
+## Conclusion
+
+Values, functions, and definitions are the most primitive and core concepts of OCaml. Even typing or execution aren't as central. The key concept is the environment. In summary, it is an ordered add-only key-value store. Add-only because nothing can be deleted. It is ordered because it is just the list of available definitions. When writing a `let`, zero, one, or several name-value pairs are added to the relevant environment. When calling a function parameter, names and arguments are also added to the environment. A closure is an environment-function pair.
```

Comment:
Let's revisit this conclusion to make it more cohesive. I'll see if I can work something up.

Reply:
Ah! This is the conclusion I wanted to work on rather than the Data Types one!